File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    var answer = 0;
    
    let q = [numbers[0], -numbers[0]];
    
    for(let i = 1; i < numbers.length; i++) {
        let newQ = [];
        while(q.length != 0) {
            let cur = q.pop();
            newQ.push(cur + numbers[i]);
            newQ.push(cur - numbers[i]);
        }
        q = newQ;
    }
    
    answer = q.reduce((count, e) => {
        if(e == target) count++;
        return count;
    }, 0);
    return answer;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    public int solution(int[] numbers, int target) {
        return dfs(0, 0, numbers, target);
    }
    
    private int dfs(int idx, int sum, int[] numbers, int target) {
        int count = 0;
        
        if(idx == numbers.length) {
            if(sum == target)
                return 1;
            else 
                return 0;
        }
        
        count += dfs(idx + 1, sum + numbers[idx], numbers, target);
        count += dfs(idx + 1, sum - numbers[idx], numbers, target);
        
        return count;
    }
}